### DISCUSSION TITLE
anvil: is there a way to retrieve the sidecar after a blob transaction has been mined?

### DISCUSSION THREAD

**Discussion by davidtaikocha** [foundry-rs/foundry#9221] [Help]
In our use case (L2), we want to fetch the sidecar blob after a blob transaction has been included, is there anyway to do so right now?

**Comment by yash-atreya** [comment]
@davidtaikocha We don't persist the blobs in anvil. Blobs are persisted on the beacon node. cc @mattsse

**Comment by davidtaikocha** [comment]
> @davidtaikocha We don't persist the blobs in anvil. Blobs are persisted on the beacon node. cc @mattsse

But we are sending blobs to anvil as EL (so without using a beacon node), which means there is noway to fetch a blob after mined?

**Comment by Thegaram** [comment]
@yash-atreya We also believe it would be useful to add an option to anvil to persist blobs and expose a subset of the beacon API to allow querying them. Many rollup teams might want to use anvil in their test/devnet setup.

**Reply by yash-atreya** [reply]
What would be your ideal workflow? Which beacon APIs should be exposed?

**Reply by yash-atreya** [reply]
Can you create an issue? And add details there?

**Reply by Thegaram** [reply]
Thanks, created an issue: https://github.com/foundry-rs/foundry/issues/10978
